snapcraft/storeapi/assertions.py

```diff
@@ -0,0 +1,174 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import json
+import subprocess
+from copy import deepcopy
+from datetime import datetime
+
+from . import StoreClient
+from . import errors
+from . import constants
+
+
+class _BaseAssertion:
+
+    @property
+    def publisher_id(self):
+        if not self._account_info:
+            self._account_info = self._store_client.get_account_information()
+        return self._account_info['account_id']
+
+    @property
+    def snap_id(self):
+        if not self._account_info:
+            self._account_info = self._store_client.get_account_information()
+        snaps = self._account_info['snaps'][self._release]
+        try:
+            return snaps[self._snap_name]['snap-id']
+        except KeyError:
+            raise errors.SnapNotFoundError(self._snap_name)
+
+    def __init__(self, *, snap_name, signing_key):
+        self._store_client = StoreClient()
+        self._snap_name = snap_name
+        self._signing_key = signing_key
+        self._release = constants.DEFAULT_SERIES
+        self._account_info = None
+        self.assertion = None
+        self.signed_assertion = None
+
+    def get(self):
+        self.assertion = self._store_client.get_assertion(
+            self.snap_id, self._assertion_type)
+        return self.assertion
+
+    def sign(self):
+        cmdline = ['snap', 'sign']
+        if self._signing_key:
+            cmdline += ['-k', self._signing_key]
+        snap_sign = subprocess.Popen(
+            cmdline, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE)
+        data = json.dumps(self.assertion).encode('utf8')
+        assertion, err = snap_sign.communicate(input=data)
+        if snap_sign.returncode != 0:
+            err = err.decode('ascii', errors='replace')
+            raise errors.StoreAssertionError(
+                snap_name=self._snap_name,
+                endpoint=self._assertion_type, error=err)
+
+        self.signed_assertion = assertion
+        return assertion
+
+    def push(self, force=False):
```

Comment:
Please add a docstring for the force arg, explaining that it can remove colaborators.

Reply:
done